pramen/core/src/main/scala/za/co/absa/pramen/core/metastore/MetastoreImpl.scala

```diff
@@ -48,7 +48,14 @@ class MetastoreImpl(appConfig: Config,
   }
 
   override def isDataAvailable(tableName: String, infoDateFromOpt: Option[LocalDate], infoDateToOpt: Option[LocalDate]): Boolean = {
-    bookkeeper.getDataChunksCount(tableName, infoDateFromOpt, infoDateToOpt) > 0
+    val mt = getTableDef(tableName)
+    val isOnDemand = mt.format.isInstanceOf[DataFormat.Transient]
+
+    if (isOnDemand) {
+      true
```

Comment:
ToDo. Actually, now that I'm looking at this line again, it can be improved. The program can check if there are any valid information dates for this period, and return true only if it is.